modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ObjectRelationship/ObjectRelationship.js

```diff
@@ -12,33 +12,101 @@
  * details.
  */
 
-import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocomplete';
+import {useIsMounted} from '@liferay/frontend-js-react-web';
 import {FieldBase} from 'dynamic-data-mapping-form-field-type/FieldBase/ReactFieldBase.es';
-import React from 'react';
+import {Autocomplete as BaseAutocomplete} from 'frontend-js-components-web';
+import {debounce, fetch} from 'frontend-js-web';
+import React, {useState} from 'react';
 
-const ObjectRelationship = ({
+function getData(apiUrl, query, page, pageSize) {
+	const url = new URL(apiUrl, themeDisplay.getPortalURL());
+
+	if (query) {
+		url.searchParams.set('search', query);
+	}
+
+	if (page) {
+		url.searchParams.set('page', page);
+	}
+
+	if (pageSize) {
+		url.searchParams.set('pageSize', pageSize);
+	}
+
+	return fetch(url, {
+		headers: new Headers({
+			Accept: 'application/json',
+			'Accept-Language': themeDisplay.getBCP47LanguageId(),
+			'Content-Type': 'application/json',
+		}),
```

Comment:
@jbalsas I am going to leave this as is for now. I looked into exposing it as a constant with `fetch` but there are a handful of use cases that need to be thought through. I created a task for this. https://issues.liferay.com/browse/LPS-139556

Reply:
Accept and Content-Type are not needed AFAIK if we want to rely on defaults.